backends/bmv2/jsonconverter.cpp

```diff
@@ -803,13 +805,34 @@ JsonConverter::JsonConverter(const CompilerOptions& options) :
         headerParameter(nullptr), userMetadataParameter(nullptr), stdMetadataParameter(nullptr)
 {}
 
+void addSrcInfoData(Util::JsonObject* result,
```

Comment:
It would be helpful to take advantage of ```SourceFileLine``` (see lib/source_file.h:221) and add a toJson(Util::JsonObject *output) method. Then you can simply ask each node to print its SourceFileLine object.